### ISSUE TITLE
CoveragePlot on integrated object?

### ISSUE THREAD

**Issue by laurabchopp** [timoast/signac#98]
Hi Tim,

Is it possible to do a CoveragePlot on an integrated object?
I have four datasets that I integrated, and I generated a consolidated fragment file as you show in your FAQ.
However, when I try to do a CoveragePlot it shows up blank.
I'm guessing this is because it can't match the cell name (which gets re-named at integration) to the fragment file.

My code is below:
Sample 1-4 are seurat objects that have been filtered and pre-processed, and have a new assay that is called "MergedPeaks", generated by counting the reads at a consensus peakset.

```
Sample1$orig.ident <- "Exp1"
Sample2$orig.ident <- "Exp1"
Sample3$orig.ident <- "Exp2"
Sample4$orig.ident <- "Exp2"

ATAC <- merge(Sample1, y = c(Sample2, Sample3, Sample4))


ATAC.list <- SplitObject(ATAC, split.by = "orig.ident")

for (i in 1:length(ATAC.list)) {
  ATAC.list[[i]] <- RunTFIDF(ATAC.list[[i]], verbose = FALSE)
  ATAC.list[[i]] <- FindTopFeatures(ATAC.list[[i]], min.cutoff = 50)
  ATAC.list[[i]] <- RunSVD(ATAC.list[[i]], reduction.key = 'LSI_', reduction.name = 'lsi')
}
reference.list <- ATAC.list[c("Exp1", "Exp2")]


atac.anchors <- FindIntegrationAnchors(object.list = reference.list, assay = c("MergedPeaks", "MergedPeaks"), anchor.features = peaks.use, k.filter=NA)
atac.integrated <- IntegrateData(anchorset = atac.anchors, weight.reduction = reference.list[['lsi']], preserve.order = TRUE)


atac.integrated <- RunSVD(
  object = atac.integrated,
  n = 30,
  reduction.name = 'integratedLSI'
)

atac.integrated <- RunUMAP(
  object = atac.integrated,
  dims = 1:30,
  reduction = 'integratedLSI'
)


DefaultAssay(atac.integrated) <- "MergedPeaks"
fragment.path <- 'CombinedFragments.tsv.gz'
atac.integrated <- SetFragments(
  object = atac.integrated,
  file = fragment.path)


CoveragePlot(
  object = atac.integrated,
  region = "chr3:37102060-37258682",
  sep = c(":", "-"),
  annotation = EnsDb.Mmusculus.v79,
  extend.upstream = 10,
  extend.downstream = 10,
  ncol = 1, idents = c("0", "1", "2", "3","4", "5", "6", "7", "8", "10"))

##Returns blank plot###
```

The code has worked in the past on non-integrated datasets.
Thanks,
Laura

**Comment by timoast** [comment]
Hi Laura, it's definitely possible you just need to ensure the cell names are consistent across the object and the fragment file.

If you run `RenameCells` for each object before integration you can add a known suffix to the cell barocdes, then you can add the same suffix to all the cell barcodes in the fragment file before merging the different fragment files. See https://github.com/timoast/signac/issues/45#issuecomment-563324805 for more info

**Comment by laurabchopp** [comment]
@timoast Thank you SO much, that worked perfectly. Very much appreciate your helpful responses. Thanks for such an amazing analysis tool!

**Comment by Nafarroa1512** [comment]
Hi @laurabchopp!
I know that the chat is closed, but I have a doubt with your script. How did you defined MergedPeaks?
I mean, what does it contains? Thank you very much for your help in advance.

**Comment by laurabchopp** [comment]
The peakset I used was generated by finding the intersecting regions from the peakset generated by CellRanger Aggr for each independent experiment. Essentially, CellRanger Aggr calls peaks on samples 1 and 2 together, and samples 3 and 4 together, then I took the intersection. Tim gives a nice example of a good way to get a merged peakset in https://github.com/timoast/signac/issues/89#issuecomment-587496087
I just used a different way because I did that step weeks ago.
